pkg/user/handler.go

```diff
@@ -265,3 +270,51 @@ func (h Handler) FindAll(c *gin.Context) {
 
 	c.JSON(http.StatusOK, users)
 }
+
+// Delete user
+func (h Handler) Delete(c *gin.Context) {
+	// swagger:route DELETE /users/{id} deleteUser
+	//
+	// Delete user
+	//
+	// Delete user by id
+	//
+	// Security:
+	//	oauth2:
+	//
+	// Responses:
+	//	202:
+	//	401: Error
+	//	403: Error
+	//	404: Error
+	//	415: Error
+	id, ok := handler.GetPathParameter(c, "id")
+	if !ok {
+		return
+	}
+
+	user, err := handler.GetUserFromContext(c)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	_, err = h.userService.FindById(uint(id))
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	if user.ID == uint(id) {
+		_ = c.AbortWithError(http.StatusBadRequest, errors.New("cannot delete the current user"))
```

Comment:
That's not what I meant 😅 

If you are fine with the status codes in https://github.com/dhis2-sre/im-manager/blob/9060e0af8ba4eaeae67309f45b7b4f29d8248ba3/internal/middleware/errorHandler.go#L27 then all you need is `_ = c.Error(err)`. So if you want HTTP status code 400 do `_ = c.Error(errdef.NewBadRequest("cannot delete the current user"))`. `errdef` IMHO contains and should only contain HTTP independent/generic errors used throughout the app. The mapping to HTTP happens in the HTTP aware error middleware `errorHandler`.

If you want to return a specific HTTP status code that we do not already map in the `errorHandler` use `_ = c.AbortWithError(http.StatusForbidden, errors.New("cannot delete the current user"))` with the status code in a handler.

Reply:
Okay, I get it now. I was slightly confused why I had to use `AbortWithError` and just went with your previous comment. 😅